lib/octopus_serverspec_extensions/type/octopus_deploy_tentacle.rb

```diff
@@ -121,6 +119,7 @@ def has_display_name?(name)
 
     def has_endpoint?(uri)
       return false if @machine.nil?
+      return false if @machine["Uri"].nil? # polling tentacles have null endpoint. catch that.
```

Comment:
This concerns me...
I wonder if we should be checking explicitly for polling.

Reply:
Is there a case where "uri" might be nil but it's not a polling tentacle? At the moment this little bit of code is just there to handle the case where you've tried to test a polling tentacle for an endpoint it doesn't have.